src/coiio/imageio.cpp

```diff
@@ -0,0 +1,201 @@
+                      stride_t ystride, stride_t zstride,
+                      ProgressCallback progress_callback,
+                      void* progress_callback_data)
+{
+    return ii->read_image(subimage, miplevel, chbegin, chend,
+                          pun<OIIO::TypeDesc>(format), data, xstride, ystride,
+                          zstride, progress_callback, progress_callback_data);
+}
+
+OIIO_API bool
+ImageInput_has_error(const ImageInput* ii)
+{
+    return ii->has_error();
+}
+
+const char*
+ImageInput_geterror(const ImageInput* ii)
+{
+    // If we have multiple ImageInputs that we want to get errors for, we're
+    // still passing through this one function. We have to cache the string
+    // in order to return a char*, so it needs to be thread_local to avoid issues.
+    // We force clear here so that multiple calls from different ImageInputs
+    // don't get each others' errors.
+    thread_local std::string errorstring;
+    errorstring = ii->geterror(true);
+    return errorstring.c_str();
+}
+
+ImageOutput*
+ImageOutput_create(const char* filename, void* ioproxy,
+                   const char* plugin_search_path)
+{
+    return ImageOutput::create(filename, nullptr, plugin_search_path).release();
+}
+
+void
+ImageOutput_delete(ImageOutput* io)
+{
+    delete io;
+}
+
+bool
+ImageOutput_open(ImageOutput* io, const char* name, const ImageSpec* newspec,
+                 int mode)
+{
+    return io->open(std::string(name), *newspec,
+                    (OIIO::ImageOutput::OpenMode)mode);
+}
+
+OIIO_API bool
+ImageOutput_has_error(const ImageOutput* io)
+{
+    return io->has_error();
+}
+
+const char*
+ImageOutput_geterror(const ImageOutput* io)
+{
+    // If we have multiple ImageOutputs that we want to get errors for, we're
+    // still passing through this one function. We have to cache the string
+    // in order to return a char*, so it needs to be thread_local to avoid issues.
+    // We force clear here so that multiple calls from different ImageOutputs
+    // don't get each others' errors.
+    thread_local std::string errorstring;
+    errorstring = io->geterror(true);
+    return errorstring.c_str();
+}
+
+bool
+ImageOutput_write_image(ImageOutput* io, TypeDesc format, const void* data,
+                        stride_t xstride, stride_t ystride, stride_t zstride,
+                        ProgressCallback progress_callback,
+                        void* progress_callback_data)
+{
+    return io->write_image(pun<OIIO::TypeDesc>(format), data, xstride, ystride,
+                           zstride, progress_callback, progress_callback_data);
+}
+int
+openimageio_version()
+{
+    return OIIO::openimageio_version();
+}
+
+bool
+openimageio_haserror()
+{
+    return OIIO::has_error();
+}
+
+const char*
+openimageio_geterror(bool clear)
+{
+    thread_local std::string errorstring;
+    errorstring = OIIO::geterror(clear);
+    return errorstring.c_str();
+}
+}
```

Comment:
I think there should be one more linefeed here

Reply:
Actually this appears to be done by clang-format. It's controlled with BreakBeforeBraces which is currently set to WebKit.

Reply:
I meant at the end of the file... usually the last character of the file is a linefeed, not the brace itself.

Reply:
Ahh gotcha.